### PR TITLE
Stock transfer show

### PR DESCRIPTION
**Pull request by allisonlarson**
Pulls out the transfer item table into a partial, with ability to show received/expected quantitys and/or add actions to those fields.

Adds a stock transfer show page.

### CODE REVIEW

File: backend/app/views/spree/admin/stock_transfers/_transfer_item_actions.html.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<% if actions %>
```
Comment (richardnuno) [inline backend/app/views/spree/admin/stock_transfers/_transfer_item_actions.html.erb]: I think something like `show_actions` might be clearer variable name

File: backend/app/views/spree/admin/stock_transfers/_transfer_item_actions.html.erb [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  <%= render partial: 'spree/admin/shared/number_field_update_cell', locals: { resource_id: item.id, field_tag: "#{quantity_type}_quantity", number_value: item.send("#{quantity_type}_quantity") } %>
+  <td class="actions">
+    <%= render partial: 'spree/admin/shared/number_field_update_actions', locals: { resource: item, update_data: {} } %>
+    <%= link_to_with_icon 'trash', Spree.t('delete'), '#', no_text: true, data: { action: 'remove', id: item.id } if can?(:destroy, item) %>
```
Comment (richardnuno) [inline backend/app/views/spree/admin/stock_transfers/_transfer_item_actions.html.erb]: Will the delete button be displayed when receiving?

File: backend/app/views/spree/admin/stock_transfers/edit.html.erb [hunk 3]
```diff
@@ -66,70 +66,5 @@
...
-            <%= link_to_with_icon 'trash', Spree.t('delete'), '#', no_text: true, data: { action: 'remove', id: item.id } if can?(:destroy, item) %>
-          </td>
-        </tr>
-      <% end %>
-    </tbody>
-  </table>
-  <div class="no-objects-found" <%= 'hidden' if @stock_transfer.transfer_items.any? %>>
-    <%= Spree.t(:no_resource_found, resource: I18n.t(:other, scope: 'activerecord.models.spree/transfer_item')) %>
-  </div>
+  <%= render partial: 'transfer_item_table', locals: { expected: true, received: false, actions: true } %>
```
Comment (richardnuno) [inline backend/app/views/spree/admin/stock_transfers/edit.html.erb]: Same with `expected` and `received`, prefixing them with `show_` or something equivalent makes the boolean values make more sense.

**Comment by richardnuno** [comment]
Looks good overall but the special permission to view the show page (and the check for that permission in the index page) is missing.

**Comment by athal7** [comment]
@richardnuno i think we get that for free with resource controller since it authorizes the showing of stock transfers

**Comment by richardnuno** [comment]
@athal7 good point, @allisonlarson no need for a special permission then

**Comment by allisonlarson** [comment]
Yep, just tested. We get that fo' free

**Comment by gmacdougall** [comment]
:+1: Looks good. Nice improvement over what was there.

**Comment by richardnuno** [comment]
:+1:

**Pull request merged**